arch/etc/initcpio/hooks/bcachefs

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/ash
+
+run_hook() {
+
+# check if $root needs unlocking
+if bcachefs unlock -c $root >/dev/null 2>&1; then
```

Comment:
This fails if `root` isn't a dev path (eg. `UUID=`), because only `bcahcefs mount` knows how to handle that.

Could do what some existing hooks do ([resume](https://github.com/archlinux/mkinitcpio/blob/e60da408c832eefd7fcb7ad315a00e18a0832fd4/hooks/resume#L31))

```
if rootdev="$(resolve_device "$root")" && \
bcachefs unlock -c $rootdev >/dev/null 2>&1; then
    echo "Unlocking $rootdev:"
    while true; do
        bcachefs unlock "$rootdev" && break
    done
fi
```